sites/all/modules/custom/lesson_entity/lesson_entity.module

```diff
@@ -0,0 +1,330 @@
+      'full' => [
+        'label' => t('Full'),
+        'custom settings' => FALSE,
+      ],
+    ],
+  ];
+
+  return $info;
+}
+
+/**
+ * Implements hook_menu().
+ */
+function lesson_entity_menu() {
+  $items['custom/lesson_entity'] = [
+    'title' => 'Lesson Entity',
+    'page callback' => 'lesson_entity_info_page',
+    'access arguments' => ['view any lesson_entity_basic entity'],
+  ];
+
+  $items['admin/structure/lesson_entity_basic/manage'] = [
+    'title' => 'Administer lesson_entity_basic entity type',
+    'page callback' => 'lesson_entity_basic_list_entities',
+    'access arguments' => ['administer lesson_entity_basic entities'],
+  ];
+
+  $items['admin/structure/lesson_entity_basic/manage/add'] = [
+    'title' => 'Add a Lesson Entity Basic Entity',
+    'page callback' => 'lesson_entity_basic_add',
+    'access arguments' => ['create lesson_entity_basic entities'],
+    'type' => MENU_LOCAL_ACTION,
+  ];
+
+  $items['admin/structure/lesson_entity_basic/manage/list'] = [
+    'title' => 'List',
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+  ];
+
+  $items['custom/lesson_entity/basic/%lesson_entity_basic'] = [
+    'page callback' => 'lesson_entity_basic_show',
+    'page arguments' => [3],
+    'access arguments' => ['view any lesson_entity_basic entity'],
+  ];
+
+  $items['custom/lesson_entity/basic/%lesson_entity_basic/view'] = [
+    'title' => 'View',
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'weight' => -10,
+  ];
+
+  $items['custom/lesson_entity/basic/%lesson_entity_basic/edit'] = [
+    'title' => 'Edit',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => ['lesson_entity_basic_form', 3],
+    'access arguments' => ['edit any lesson_entity_basic entity'],
+    'type' => MENU_LOCAL_TASK,
+  ];
+
+  $items['custom/lesson_entity/basic/add'] = [
+    'title' => 'Add a Lesson Entity Basic Entity',
+    'page callback' => 'lesson_entity_basic_add',
+    'access arguments' => ['create lesson_entity_basic entities'],
+  ];
+
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function lesson_entity_permission() {
+  $permissions = [
+    'administer lesson_entity_basic entities' => [
+      'title' => t('Administer lesson_entity_basic entities'),
+    ],
+    'view any lesson_entity_basic entity' => [
+      'title' => t('View any Lesson Entity Basic entity'),
+    ],
+    'edit any lesson_entity_basic entity' => [
+      'title' => t('Edit any Lesson Entity Basic entity'),
+    ],
+    'create lesson_entity_basic entities' => [
+      'title' => t('Create Lesson Entity Basic Entities'),
+    ],
+  ];
+
+  return $permissions;
+}
+
+/**
+ * Implements hook_node_access().
+ */
+function lesson_entity_node_access($node,$op,$account) {
+  if ($node->type === 'article') {
+    $result = db_select('lesson_entity','le')
+      ->fields('le', array('uid'))
+      ->condition('le.uid', $account->uid)
+      ->execute();
+
+    return ($result->rowCount()>0) ? NODE_ACCESS_ALLOW : NODE_ACCESS_DENY;

```

Comment:
It would be better to rewrite function as:
$access = NODE_ACCESS_ALLOW;
if () {
...
$access = NODE_ACCESS_DENY;
}

return $access;
It looks more clear